### ISSUE TITLE
Pydantic model objects evaluate to None when used in group_data files

### ISSUE THREAD

**Issue by relaxdiego** [Fizzadar/pyinfra#496]
**Describe the bug**
When using an object that inherits from pydantic's BaseModel as host data, it evaluates to None.

**To Reproduce**

```python
# group_data/server.py
from pydantic import BaseModel


class DhcpData(BaseModel):
    shared_network_name: str


dhcp = DhcpData(shared_network_name='pxe.lan')
```

```python
# inventories/pxe.py
server = ['kvm-1']
```

```
# templates/configure.j2
{{host.data.dhcp.shared_network_name}}
```

```python
# operations.py
from pyinfra import host
from pyinfra.operations import files

assert host.data.dhcp is not None
assert host.data.dhcp.shared_network_name == 'pxe.lan'

files.template(
    name="Test",
    src='templates/configure.j2',
    dest='configure.sh',
    mode='700',
)
```

```
$ pyinfra inventories/pxe.py operations.py
```

**Expected behavior**
`host.data.dhcp` would be available for use in `operations.py` or in any jinja template in the project directory.

**Meta**
+ Include output of `pyinfra --support`.

```
System: Darwin
  Platform: Darwin-19.6.0-x86_64-i386-64bit
  Release: 19.6.0
  Machine: x86_64
pyinfra: v1.2.1
Executable: /Users/mark/.pyenv/versions/aircraft/bin/pyinfra
Python: 3.7.7 (CPython, Clang 11.0.0 (clang-1100.0.33.17))
```

+ How was pyinfra installed (source/pip)?

pip

+ debug.log

```
  File "/Users/mark/.pyenv/versions/3.7.7/envs/aircraft/lib/python3.7/site-packages/pyinfra_cli/main.py", line 216, in cli
    _main(*args, **kwargs)
  File "/Users/mark/.pyenv/versions/3.7.7/envs/aircraft/lib/python3.7/site-packages/pyinfra_cli/main.py", line 543, in _main
    load_deploy_file(state, filename)
  File "/Users/mark/.pyenv/versions/3.7.7/envs/aircraft/lib/python3.7/site-packages/pyinfra_cli/util.py", line 168, in load_deploy_file
    exec_file(filename)
  File "/Users/mark/.pyenv/versions/3.7.7/envs/aircraft/lib/python3.7/site-packages/pyinfra_cli/util.py", line 52, in exec_file
    exec(PYTHON_CODES[filename], data)
  File "operations.py", line 4, in <module>
    assert host.data.dhcp is not None
AssertionError
```

+ Consider including output with `-vv` and `--debug`.

```
    [pyinfra_cli.main] Checking potential directory:
    [pyinfra_cli.main] Setting directory to:
--> Loading config...
--> Loading inventory...
    [pyinfra_cli.inventory] Creating fake inventory...
    [pyinfra_cli.inventory] Looking for group data in: group_data/server.py

--> Connecting to hosts...
    [pyinfra.api.connectors.ssh] Connecting to: kvm-1 ({'allow_agent': True, 'look_for_keys': True, 'hostname': 'kvm-1', 'timeout': 10})
    [kvm-1] Connected
    [pyinfra.api.state] Activating host: kvm-1

--> Preparing operations...
    Loading: operations.py
--> An unexpected exception occurred:

  File "operations.py", line 4, in <module>
    assert host.data.dhcp is not None
AssertionError
    [pyinfra_cli.exceptions]   File "/Users/mark/.pyenv/versions/3.7.7/envs/aircraft/lib/python3.7/site-packages/pyinfra_cli/main.py", line 216, in cli
    _main(*args, **kwargs)
  File "/Users/mark/.pyenv/versions/3.7.7/envs/aircraft/lib/python3.7/site-packages/pyinfra_cli/main.py", line 543, in _main
    load_deploy_file(state, filename)
  File "/Users/mark/.pyenv/versions/3.7.7/envs/aircraft/lib/python3.7/site-packages/pyinfra_cli/util.py", line 168, in load_deploy_file
    exec_file(filename)
  File "/Users/mark/.pyenv/versions/3.7.7/envs/aircraft/lib/python3.7/site-packages/pyinfra_cli/util.py", line 52, in exec_file
    exec(PYTHON_CODES[filename], data)
  File "operations.py", line 4, in <module>
    assert host.data.dhcp is not None

    [pyinfra_cli.exceptions] AssertionError


--> The full traceback has been written to pyinfra-debug.log
```

**Comment by Fizzadar** [comment]
Hi @relaxdiego! The reason for this is due to the limited types pyinfra accepts as group data (see: https://github.com/Fizzadar/pyinfra/blob/master/pyinfra_cli/inventory.py#L21-L25).

However, I don't believe there's any reason not to allow any types as data. Going to double check on that but I like the idea of annoying anything to be defined as group data.

In the meantime an immediate workaround for your usecase could be to put the `dhcp` variable within a dictionary object which will become data.

**Comment by relaxdiego** [comment]
Thanks @Fizzadar. Pydantic does offer a `dict()` method (e.g. `dhcp = DhcpData(shared_network_name='pxe.lan').dict()` which is the workaround that I use today. However, that loses the dot notation goodness of accessing its members within the code which becomes inconsistent with the retained dot notation if the Jinja templates.

**Comment by relaxdiego** [comment]
I also realized another hacky workaround which is to monkey patch `ALLOWED_DATA_TYPES` by adding this to the top of the inventory:

```python
from pydantic import BaseModel
from pyinfra_cli import inventory

inventory.ALLOWED_DATA_TYPES = tuple(inventory.ALLOWED_DATA_TYPES + (BaseModel,))
```

> WARNING: Don't try this at home (read: production), kids!

**Comment by Fizzadar** [comment]
So I've looked into this - this limitation dates back years now and is no longer relevant so I've removed it in https://github.com/Fizzadar/pyinfra/commit/48e8a1e99684ea35ca99054ae5ff8459b8c87339. Anything goes in data files now!

**Comment by Fizzadar** [comment]
(will release this in v1.3.1 today/this week).

**Comment by Fizzadar** [comment]
Now released in `1.3.1`!
